curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md

```diff
@@ -47,17 +47,20 @@ You should call your `detectFullHouse` function when your `rollDiceBtn` is click
 ```js
 const counts = (arr) => arr.reduce((counts, num) => {counts[num] = ++counts[num] || 1; return counts}, {});
 const isFullHouse = (arr) => arr.includes(3) && arr.includes(2) && arr.length === 2;
+let foundFullHouse = false;
 // Absurdly high, but will either timeout or bail early when assertion runs.
 for (let i = 0; i < 100_000; i++) {
   rolls = 0;
   rollDiceBtn.click();
-  if (isFullHouse(diceValuesArr)) {
+  if (isFullHouse(Object.values(counts(diceValuesArr)))) {
     assert.isFalse(scoreInputs[2].disabled);
     assert.strictEqual(scoreInputs[2].value, "25");
     assert.strictEqual(scoreSpans[2].innerText, ", score = 25");
+    foundFullHouse = true;
     break;
   }
 }
+assert.isTrue(foundFullHouse);
```

Comment:
Why can't we override the user's the roll button behavior to instead set a dice roll that we want so we don't have to set up this huge loop? We then could test against a couple possible full house results.

Reply:
Mainly because I was unable to find any documentation about overriding ```const``` variables in tests. In the end, it is clear to me that there is a transcompiling process that converts user variables from ```const``` to ```var``` which has the side effect of allowing tests to override them. However, I am new to this codebase, so it is unclear to me if this is a desirable practice.

Reply:
You don't need to override a const variable to do what I proposed.

Reply:
The purpose of this test is to check if the camper has called the detectFullHouse function in their implementation of the on click event handler. detectFullHouse is already being directly tested to see if it exists and works as intended.

Reply:
But this step already has a version of the onClick set in the starting code

Reply:
users are modifying the onClick function given in the seed in this step. They do not modify the onClick in every step, but they are directed to in this step.

I see two main options for removing the looping.

1. Temporarily override ```Math.random``` to create well-defined behavior in ```roleDice```. I have created a PR with this method and am waiting for it to pass checks: #55499 
2. Take advantage of the transcompiling process to override the otherwise ```const``` function ```roleDice``` to create well-defined behavior. This is the method advocated in #55471

Reply:
Now I'm tracking. I'd override the RNG, personally